sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1883,7 +1883,19 @@ class Analyzer(
       // Second, we group extractedWindowExprBuffer based on their Partition and Order Specs.
       val groupedWindowExpressions = extractedWindowExprBuffer.groupBy { expr =>
         val distinctWindowSpec = expr.collect {
-          case window: WindowExpression => window.windowSpec
+          case window: WindowExpression =>
+            val winExpr = window.windowFunction
+            val distinctOpt = winExpr.find (expr => expr.isInstanceOf[AggregateExpression]
+                && expr.asInstanceOf[AggregateExpression].isDistinct)
+            if (distinctOpt.nonEmpty && window.windowSpec.orderSpec.nonEmpty) {
```

Comment:
So this only works for global window frames? If it does why did you implement distinct processing for growing, sliding and shrinking frames?